api/providers/views.py

```diff
@@ -619,3 +637,108 @@ def get_default_queryset(self):
 
     def get_queryset(self):
         return self.get_queryset_from_request()
+
+
+class RegistrationProviderRegistrationList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
+    """The documentation for this endpoint can be found [here](https://developer.osf.io/#operation/preprint_providers_preprints_list).
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
+    )
+
+    ordering = ('-created')
+
+    serializer_class = DraftRegistrationSerializer
+
+    required_read_scopes = [CoreScopes.NODE_REGISTRATIONS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    view_category = 'registration-providers'
+    view_name = 'registrations-list'
+
+    def get_provider(self):
+        # used in perms class
+        return get_object_or_error(
+            RegistrationProvider,
+            self.kwargs['provider_id'],
+            self.request,
+            display_name='RegistrationProvider',
+        )
+
+    def get_default_queryset(self):
+        return DraftRegistration.objects.filter(
+            provider=self.get_provider(),
+            registered_node__registration_approval__state=RegistrationApproval.APPROVED,
+        )
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+
+class RegistrationProviderRequestList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        base_permissions.TokenHasScope,
+        MustBeModerator,
+    )
+    view_category = 'requests'
+    view_name = 'registration-provider-withdrawal-request-list'
+
+    required_read_scopes = [CoreScopes.REGISTRATION_REQUESTS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    serializer_class = RegistrationRequestSerializer
+
+    def get_provider(self):
+        # used in perms class
+        return get_object_or_error(
+            RegistrationProvider,
+            self.kwargs['provider_id'],
+            self.request,
+            display_name='RegistrationProvider',
+        )
+
+    def get_default_queryset(self):
+        return NodeRequest.objects.filter(
```

Comment:
Why should we use `NodeRequest` as the object here? Is it fine that the associated machine class for the object is the NodeRequestMachine?

Reply:
This kind of just weird because NodeRequest has it's own machine_class, but it really shouldn't change or effect this too much. The idea here is that the front-end is going to be able to see all the request, but basically only needs the withdrawal ones, and it can just filter for them.